project_template/test/test_operations.cpp

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#include <catch2/catch.hpp>

```

Comment:
>catch2 [](start = 10, length = 6)

Is the plan to enforce the test harness/mocking platform? Catch2 is a C++ test runner. It does not even officially support C:

Catch2 is a multi-paradigm test framework for C++. which also supports Objective-C (and maybe C). It is primarily distributed as a single header file, although certain extensions may require additional headers. ([https://github.com/catchorg/Catch2#whats-the-catch](https://github.com/catchorg/Catch2#whats-the-catch)).

In IoT, we have a custom test runner for C (released as open source in GitHub) which is very slim (meaning that we can port to a constrained device and run tests there): [https://github.com/Azure/azure-ctest](https://github.com/Azure/azure-ctest). We also have a mocking framework ([https://github.com/Azure/umock-c](https://github.com/Azure/umock-c)).